### ISSUE TITLE
Licensing issue... can we get a better license?

### ISSUE THREAD

**Issue by protosam** [opensource-socialnetwork/opensource-socialnetwork#1445]
I tested this code out and I gotta say, it's some good work. Good job.

However, I can't use it at all. The current license is vague and very anti-user. Even though you're calling this "Open Source" Social Network, it's just not Open Source. This is because of the licensing.

The license is also just unenforceable in many aspects, due to how vague it is. I saw issue #1356, and I think your logic is bad not to use an already existing open source license. People who use open source software already know the rules on these existing licenses.

You're losing a lot of protection for your rights as creators as well as preventing people from wanting to adopt your software.

Even considering, maybe you can even use a Creative Commons license https://creativecommons.org/
They have made it very comprehensive.

At the very least, I would recommend having a real lawyer familiar with copyright law to draft you a proper license outlining limitations of liability, ownership of code, rules about how the code can be used, what is not allowed to be done, and arbitration. 

As-is:

- If I were a bad actor and decided to monetize the software, and violate the rules outlined, it would be a losing battle from your end. Depending on my locale the entire contract could be null and void.
- If I accrue damages, your liability limitations has some issues that a proper lawyer can fix.
- Pro-tip, there's an inherent issue section 2.3 that a lawyer would spot and fix for you.
- Second 2.4 also has issues too. It is vague enough to mean that I can't even remove it from the rendered page and it also conflicts with section 2.2.
- Also again, a lawyer should spot this... however there are multiple ways that I can think of, in which I can follow the rules of your pseudo-license and end up with a copy of the code that is no longer subject to the license.
- The license is very one sided in regards to guarantees. I have no protection if I use this software.

In summary, with a good contract, you will have better protection in regards to liability, enforcement, and ensuring that you have any control over your source code.

I really hope you take this into consideration and look forward to seeing some licensing improvement. The software looks pretty decent and I can tell it was made with some care. Licensing requires just as much consideration and care too. It not only protects you, but it protects the users of your software.

**Comment by lianglee** [comment]
@protosam,  i think i agree with you for the license improvement but i am not agree with anyone to use existing open source licenses.

Further the goal of the license is that you can not remove the copyrights, powered by notices that is all.   License allow you to resell. further  all the data,  users its yours.   

Yes indeed 2.4 have issue , it means actually that the logo image (not the text) can not be used or sold.  I'll submit for changes.  

You can remove word open source from pages, text etc except the copyright and the powered by, is it make sense now regarding 2.4?

**Comment by protosam** [comment]
Sounds like this isn't an Open Source software considering that. It's fine and all, but the name is misleading. 

I think your concerns are misplaced really. You're aiming to spread the name by making restrictions to what can be done with the code. However, what you're not seeing is that there's a large market you're not tapping into the potential of. All the people that run exclusively open source code. We care about licenses for what we run.

Support is where the money is really at and the people going to a site with "Powered by XYZ" means nothing to the end users. They aren't your target audience for generating revenue. You want as many people running this as possible so that you get to provide support services and have people curious about the non-free version.

If you actually open sourced the software, more people will use it. The more people that use it, the more people that will need support for it. You sell those support services and that's where good profits tend to be in my experience. Also rename "Premium" to "Enterprise" and call this the "Community Edition". Those words have had better success at selling.

Anyways, your license isn't going to be enforceable without actually fixing it.

Good luck.

**Comment by lianglee** [comment]
The license that suites i believe is Attribution Assurance License but we are still looking forward for additional OSI approved licenses

**Comment by lianglee** [comment]
License changed for OSSN v5.3 to https://opensource.org/licenses/CAL-1.0
